packages/tax/src/models/tax-rate.ts

```diff
@@ -56,17 +62,20 @@ export default class TaxRate {
   @Property({ columnType: "bool", default: false })
   is_combinable = false
 
-  @Property({ columnType: "text" })
-  tax_region_id: string
-
   @ManyToOne(() => TaxRegion, {
     fieldName: "tax_region_id",
     index: taxRegionIdIndexName,
-    cascade: [Cascade.REMOVE, Cascade.PERSIST],
+    mapToPk: true,
+    cascade: [Cascade.REMOVE],
```

Comment:
It's the other way around - when we remove a TaxRegion, its associated TaxRates should be removed - that's what this cascade ensures.

I thought about why the "soft-remove" doesn't implement it this way around also?

Reply:
just found out that those cascade are application level cascade, and since we do not populate the relations and use native delete, the cascade is not performed 🤔